### ISSUE TITLE
Improve task workflow

### ISSUE THREAD

**Issue by benhylau** [hyphacoop/organizing#140]
## Shared Description

:speaking_head: **Loomio:** N/A  
:date: **Due date:** N/A  
:dart: **Success criteria:** Develop unambiguous shared practice to propose, triage, complete, and close tasks  

You may have experienced the following scenarios:

- I see a problem but don't know how the steps to propose it so it gets discussed and resolved
- What priority should I assign to this ticket I created
- An issue belongs to a working group but no one knows who is doing it
- No one knows reasonable expectations of task completion from a person / working group
- When making a PR who should I add as a Reviewer?
- A PR is "done" but the owner doesn't know whether they have permission to merge it since they don't know whether people have looked at it
- There are so many PRs which ones should I look at in my limited time
- When there are opposing opinions on a PR we don't know how to move forward and things get held up indefinitely
- An issue is being discussed extensively at all hands but I really don't care about it, there are other issues I want to discuss
- Who should close this task that seems to be completed

It is a large scope to resolve these issues, but now that we have a better idea of our comm tools and people's sync time availabilities, I think we can improve the process of task assignment and decision making.

## To Do

- [x] **all** Discuss the problems I have outlined above
- [x] @hyphacoop/governance-wg Discuss how to effectively use sync (calls) and async time (comm tools)
(Previous TODOs:
- ~Propose potential workflows and discuss viability~
- ~Vote on Loomio to adopt a workflow~
- ~Agree on adopting the workflow~
- ~Document this shared practice in our handbook~

**Comment by benhylau** [comment]
Here are some ideas that I think could apply to us after consulting people with experience in other organizations:

**Triage meeting**

A weekly meeting where everyone can optionally participate. The purpose is to **assign** new tasks to a working group. This means we have collectively delegated the task to a working group, and its members have full authority to prioritize, complete, and close these tasks.

Of course in completing tasks, working groups may consult other members, but this is left to their discretion. We trust members of the working group to make the right calls, and are comfortable to let them make mistakes.

**Working group meetings**

Schedule four (one for each working group) 45 min weekly non-overlapping time blocks. This a mandatory meeting attended **only** by members of a working group.

Members look at the queue of tasks assigned to the working group and assign them to individual members for completion.

- Tasks that require extensive discussion can happen according to an arrangement decided by this group
- Completed tasks are closed off and will be announced at the next All Hands meeting
- Tasks that require consultation with the larger group are tagged with `discuss` for the next All Hands meeting

**All Hands meeting**

Bi-weekly 1.5 h meetings where working groups announce tasks completed and then look at the `discuss` prioritized. The prioritization is done **prior** to the meeting, async by individual members by voting on tasks tagged with `discuss`.

This is a meeting aimed to move through the queue and unblock working groups so they can move ahead on these `discuss` tasks in the next cycle, and also through the announcements quickly get a overview of where the org is at in all domains.

**Comment by patcon** [comment]
Thanks Ben! This is a useful convo, but feeling some of your solutions assume FT units. I'm unsure regularly scheduled WG meetings suit us yet

**Comment by patcon** [comment]
Thanks Ben! This is a useful convo.

- :heart: I'd prefer we found our ways by trying them, and through doing. E.g., if you want new things in all-hands process, convince next cultivator to try, or cultivate yourself and try it. Or if you want a WG process, introduce it through one working group you're on, then share what worked with others. No need to pre-define it all in lockstep upfront, right?

**Comment by benhylau** [comment]
I would recommend that we have a discussion together and get high level alignment before different groups moving ahead to trial different things. I think cultivators will be more inclined to trial something if there is broad intention to adopt this. I am happy to cultivate in Dec/Jan and trial the result of this discussion, but only if other members seem to see this as an improvement over current processes.

**Comment by benhylau** [comment]
Interesting to note that in thinking about the coordination practices, the [teams](https://github.com/orgs/hyphacoop/teams) do reflect how actively ppl are participating in the coop. I feel it's nice to use as a basis for people to decide how much their involvement based on their capacity. Basically, joining a WG means you have more say in how the coop is run, and your responsibilities are responding to **Mentions** and completing **Assigns**. If it feels overwhelming, you can leave a WG during a period, and commit less capacity to the coop.

The only recurring _everyone_ time is **All Hands** and @hyphacoop/member-workers for org level matters for members who are feeling really busy on other life things.

**Comment by patcon** [comment]
clayton from @agaric shared their wiki page for project management:
https://gitlab.com/agaric/sites/agaric-com/wikis/Project%20Management

**Comment by dcwalk** [comment]
This has been discussed in a few meetings, @benhylau drafted a document and brought it to an (underwhelming) loomio vote. I'm wondering how we see proceeding on this?

I think as of Nov (?) meeting @ben was going to step back from this. Could we reconsider the framing of this issue? Also whether folks are assigned to it? (Trying to clear stuff up before year-end

**Comment by benhylau** [comment]
I suggest we put it on the project board, make it low priority and in Backlog for now.

**Comment by dcwalk** [comment]
Can we make it actionable before just dumping it there? Or mark it blocked if that is an accurate acct? We've moved halfway through process but it seems stalled.

I'm sort of against knowingly leaving a ticket in an inaccurate state.

**Comment by benhylau** [comment]
I think the problems that motivated this issue remain unresolved, and either @hyphacoop/governance-wg or @hyphacoop/infra-ops-wg seem to be the right WG to drive the effort to resolves this.

The TODO has items that seem actionable to me, for example, WG to "propose potential workflows and discuss viability". We just don't have a WG assigned to take charge at the moment. Perhaps we can assign one, and let the WG decide if this needs re-scope or how to best address?

**Comment by patcon** [comment]
> Perhaps we can assign one, and let the WG decide if this needs re-scope or how to best address?

:+1: My vote is for governance WG ownership.

I'd personally wish to see each WG experiment with what works for them, then to share practices that feel successful. i don't feel that's the sort of conversation/direction that Infra & Ops WG understands as its strength (as was discussed in the last all-hands of the year).

**Comment by dcwalk** [comment]
Cleaning up ticket to reflect this

**Comment by benhylau** [comment]
Feel free to delete/modify previous TODOs if @hyphacoop/governance-wg feels no longer appropriate.

**Comment by patcon** [comment]
@hyphacoop/governance-wg would like discussion on this ticket at it's next meeting

**Comment by dcwalk** [comment]
Dropping this radical idea here:
- We delete week old chat history (committent to move important stuff to github)
- Reframe our comms away from chat and into more longerfoom async with our loomio and github.

h/t Common Knowledge https://commonknowledge.coop/writing/organising-online--thinking-about-chat and Basecamp: https://basecamp.com/guides/group-chat-problems

**Comment by dcwalk** [comment]
This has been lingering for something like 16 months. The nature of the problem has changed and we are proposing many improvements to task workflow as part of the Call me Chrysalis initiative: https://github.com/hyphacoop/organizing/milestone/2

Closing here for tracking in micro issues in that initiative!
